orttraining/orttraining/core/optimizer/localized_recompute.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/graph/graph_utils.h"
+#include "orttraining/core/optimizer/localized_recompute.h"
+
+using namespace ONNX_NAMESPACE;
+
+namespace onnxruntime {
+
+bool GeluRecompute::SatisfyCondition(const Graph& /*graph*/, const Node& node, const logging::Logger& /*logger*/) const {
+  const auto next_node = node.OutputNodesBegin();
+  if (next_node != node.OutputNodesEnd() && next_node->OpType() == "MatMul") {
```

Comment:
>&& next_node->OpType() == "MatMul" [](start = 41, length = 34)

why don't always recompute Gelu? If there's saving for MatMul, there should be savings in other places.